util/oidc/oidc.go

```diff
@@ -727,7 +727,7 @@ func (a *ClientApp) GetUserInfo(actualClaims jwt.MapClaims, issuerURL, userInfoP
 	// but first let's determine the expiry of the cache
 	var cacheExpiry time.Duration
 	settingExpiry := a.settings.UserInfoCacheExpiration()
-	tokenExpiry := getTokenExpiration(claims)
+	tokenExpiry := getTokenExpiration(actualClaims)
```

Comment:
This is sort of a second bugfix for the original PR. We looked at the wrong claims to find the expiry and thus always relied on settingsExpiry...